api/app/hfi/pdf_data_formatter.py

```diff
@@ -140,16 +140,21 @@ def get_merged_station_data(
     Returns all the weather station and daily data we have for a station
     """
     all_station_pdf_data: List[StationPDFData] = []
+    # We do a bunch of null checks here because station_dict is built from
+    # stations we retrieve from WF1, while the dailies come from the HFI request
+    # which can have an different set of stations
     for daily in dailies:
-        station_data = station_dict[daily.code]
+        station_data = station_dict.get(daily.code, None)
         daily_dict = daily.dict()
-        daily_dict.update(station_data)
+        if station_data is not None:
```

Comment:
There are two null checks for station_data. Is it possible to consolidate?

Reply:
Perhaps with 
```
if station_data is None:
    continue
```
at the first ocurrence of the null check and then remove the second null check?

Reply:
Good call, done in: https://github.com/bcgov/wps/pull/2838/commits/2b7eae72cc6d9b22bff392ade68cae24efcf5b5f